web/src/unread.js

```diff
@@ -864,9 +876,11 @@ export function initialize(params) {
     unread_pm_counter.set_huddles(unread_msgs.huddles);
     unread_pm_counter.set_pms(unread_msgs.pms);
     unread_topic_counter.set_streams(unread_msgs.streams);
-
     for (const message_id of unread_msgs.mentions) {
         unread_mentions_counter.add(message_id);
+        if (unread_msgs.pms.some((pm) => pm.unread_message_ids.includes(message_id))) {
+            mentions_in_direct_message.add(message_id);
+        }
```

Comment:
Using `unread_msgs.pms` for this is incorrect as it doesn't include huddles. Let's use `unread_direct_message_counter.get_msg_ids()`.

Also, a reproducer for this bug in commit description or PR description would have been nice.

Reply:
updated the PR also added the reproducer in the PR description.